### PR TITLE
Add support for skipping tests

### LINKED ISSUE #150: adapt tests to less capable libcurl versions
- trurl can get **built** with a restricted libcurl
- trurl might **execute** with a restricted libcurl

(where "restricted" means it might not have all the features or might suffer from a known bug.)

These needs to be taken into account when running tests, since not all features can evaluate and be performed in all situations.

### PR DESCRIPTION
**Pull request by krishean**
This should fix #150, but may need more attributes added to the json.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -89,47 +93,98 @@ def printConcise(self):
...
+    # directory, using absolute paths to the executable and json
+    # file makes it reliably find the executable
+    baseCmd = path.join(getcwd(), PROGNAME)
+    # the .exe on the end is necessary when using absolute paths
+    if sys.platform == "win32" or sys.platform == "cygwin":
+        baseCmd += ".exe"
+    # check if the trurl executable exists
+    if path.isfile(baseCmd):
+        # get the run-time and build-time libcurl versions
+        output = run(
```
[the remark this answers is not in the document]
Comment (krishean) [reply test.py:111]: @sonatype-lift ignore

**Comment by krishean** [comment]
Alright why does the github documentation say to use "python" in the yaml if it's going to fail, and why does macos even recognize that python 2 exists at this point. On ubuntu there's a "python-is-python3" package that solves this

**Comment by krishean** [comment]
It's not showing me what the conflicts are and trying to fix it seems to just be making things worse.

**Comment by bagder** [comment]
I think that's just another reason to not use the GitHub UI for this. If you use plain git and rebase this PR on top of orgin/master, `test.py` shows a significant conflict.

**Comment by krishean** [comment]
I think I fixed it, after fighting with git for awhile. Also it should be noted that using shlex.join() in test.py requires a minimum of Python 3.8

**Comment by bagder** [comment]
> unsure if it's run-time or build-time

I created #168 to help determining things like this.

**Comment by bagder** [comment]
Thanks!

**Pull request merged**